content/learn/book/ecs/resources/_index.md

```diff
@@ -6,12 +6,296 @@ page_template = "book-section.html"
+For advice on how to work with the [`World`] exposed by the [`FromWorld::from_world`] method, see the section on [exclusive world access](../exclusive-world-access/_index.md).
+
+[`FromWorld::from_world`]: https://docs.rs/bevy/latest/bevy/ecs/world/trait.FromWorld.html#tymethod.from_world
+
+## Optional resources
+
+Sometimes, you may need to write a system that is robust to whether or not a resource exists.
+We can handle both the case where it exists and the case where it doesn't by requesting `Option<Res<T>>` (or other resource types like [`ResMut`], [`NonSend`] and [`NonSendMut`]) as a system parameter,
+and then branching on the resulting `Option<T>` returned.
+
+Here's a quick runnable example:
+
+```rust
+use bevy::prelude::*;
+
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_system(countdown.system())
+        .run()
+}
+
+struct Countdown {
+    time_remaining: u8,
+}
+
+// `countdown` does not need to be marked as `mut` here,
+// as destructuring (like via `match`) does not require mutation
+// Only the internal data (`validated_countdown`) needs to be marked as `mut`
+fn countdown(
+    countdown: Option<ResMut<Countdown>>,
+    mut commands: Commands,
+    // Under the hood, this data is stored in a ResMut<Events<AppExit>>
+    mut app_exit: EventWriter<AppExit>,
+) {
+    match countdown {
+        // If no Countdown resource exists, add one
+        None => commands.insert_resource(Countdown { time_remaining: 10 }),
+        // If the resource exists, count it down
+        Some(mut validated_countdown) => {
+            info!("{} ticks remaining!", validated_countdown.time_remaining);
+            if validated_countdown.time_remaining > 1 {
+                validated_countdown.time_remaining -= 1;
+            } else {
+                // When we reach zero, exit the App
+                info!("Ka-BOOM!");
+                app_exit.send(AppExit);
+            }
+        }
+    }
+}
+```
+
+[`NonSendMut]: https://docs.rs/bevy/latest/bevy/ecs/system/struct.NonSendMut.html
+
+## `NonSend` resources
+
+Non-send resources are used to store data that do not meet the [`Send + Sync`] trait bounds: this data cannot be sent safely across threads.
+Their use cases are typically quite advanced and tend to involve interfacing with external libraries for things like audio or networking.
+`NonSend<R>` and `NonSendMut<R>` can be directly substituted for `Res<R>` and `ResMut<R>` in any system.
+The inclusion of one or more non-send resources in your system will force that system to run on the main thread,
+rather than being automatically scheduled to the first available thread.
+This can negatively impact system-level parallelism: systems with any nonsend parameters are always incompatible with each other as they both need to run on the main thread.
+
+[`Send + Sync`]: https://doc.rust-lang.org/nomicon/send-and-sync.html
```

Comment:
I'm pretty sure markdownlint is only failing because this is missing a new line at the end of the file. I believe you can configure vscode to do that automatically.